eslint.config.js

```diff
@@ -0,0 +1,32 @@
+const { FlatCompat } = require('@eslint/eslintrc');
+
+const compat = new FlatCompat({
+	baseDirectory: __dirname,
+});
+module.exports = [
+	...require('@balena/lint/config/eslint.config'),
+	...compat.config({
+		parserOptions: {
+			project: 'tsconfig.dev.json',
+		},
+		ignorePatterns: ['**/generate-completion.js', '**/bin/**/*'],
+		rules: {
+			ignoreDefinitionFiles: 0,
+			'@typescript-eslint/no-non-null-assertion': 'off',
+			'@typescript-eslint/no-shadow': 'off',
+			'@typescript-eslint/no-var-requires': 'off',
+			'@typescript-eslint/no-require-imports': 'off',
+			'@typescript-eslint/no-unnecessary-type-assertion': 'off',
+			'@typescript-eslint/prefer-nullish-coalescing': 'warn',
```

Comment:
Why do we need these? I'm okay with keeping these two but I think we can just remove the others
```suggestion
			'@typescript-eslint/no-var-requires': 'off',
			'@typescript-eslint/no-require-imports': 'off',
```

Reply:
> warn

I would prefer to address everything for that matter, warnings included

Reply:
@myarmolinsky Yes, I would prefer address everything but nullish-coalescing would be huge and potential to breaking a lot of stuff. I would rather have them as a warning for now and migrate step by step. The others are historical reasons, I didn't add them they just the ones currently used for the CLI. The only one I added was `'@typescript-eslint/no-require-imports'` for obvious reasons.